code/modules/clothing/under/accessories/accessory.dm

```diff
@@ -37,8 +37,7 @@
 			var/mob/M = has_suit.loc
 			A.Grant(M)
 
-	for(var/armor_type in armor)
```

Comment:
TG has:

```
	if (islist(U.armor) || isnull(U.armor)) 										// This proc can run before /obj/Initialize has run for U and src,
		U.armor = getArmor(arglist(U.armor))	// we have to check that the armor list has been transformed into a datum before we try to call a proc on it
																					// This is safe to do as /obj/Initialize only handles setting up the datum if actually needed.
	if (islist(armor) || isnull(armor))
		armor = getArmor(arglist(armor))
```

TG has this to prevent runtimes in a few rare cases---we should *probably* do the same in case we have items that are generated or equipped at odd points.

Reply:
I hadn't thought about that possibility, thanks for pointing it out!